### PR TITLE
[Attributes][PoC] Non translatable attribute values - New approach

### PR DESCRIPTION
**Pull request by Tomanhez**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

Adding `noTranslatable` attributes and improving attributes management on products need new approach on getting data, previous approach was very complicated. I think that getting productAttributes json and create forms and rest details on frontend will be better solutions than present magic generating forms and returning `html`.

```
{
  "attributes": {
    "jeans_brand": {
      "code": "jeans_brand",
      "type": "text",
      "translatable": true,
      "values": [
        {
          "id": "937",
          "value": "You are breathtaking",
          "label": "Jeans brand",
          "localeCode": "en_US"
        },
        {
          "id": "938",
          "value": "You are breathtaking",
          "label": "Jeans brand",
          "localeCode": "de_DE"
        },
        {
          "id": "939",
          "value": "You are breathtaking",
          "label": "Jeans brand",
          "localeCode": "fr_FR"
        },
        {
          "id": "940",
          "value": "You are breathtaking",
          "label": "Jeans brand",
          "localeCode": "pl_PL"
        }
      ]
    },
    "jeans_collection": {
      "code": "jeans_collection",
      "type": "text",
      "translatable": true,
      "values": [
        {
          "id": "945",
          "value": "Sylius Winter 2019",
          "label": "Jeans collection",
          "localeCode": "en_US"
        },
        {
          "id": "946",
          "value": "Sylius Winter 2019",
          "label": "Jeans collection",
          "localeCode": "de_DE"
        },
        {
          "id": "947",
          "value": "Sylius Winter 2019",
          "label": "Jeans collection",
          "localeCode": "fr_FR"
        },
        {
          "id": "948",
          "value": "Sylius Winter 2019",
          "label": "Jeans collection",
          "localeCode": "pl_PL"
        }
      ]
    },
    "jeans_material": {
      "code": "jeans_material",
      "type": "text",
      "translatable": false,
      "values": [
        {
          "id": "953",
          "value": "100% jeans",
          "label": "Jeans material"
        }
      ]
    }
  }
}
```

### CODE REVIEW

File: src/Sylius/Bundle/ProductBundle/Controller/ProductAttributeController.php [hunk 1]
```diff
@@ -79,6 +85,58 @@ public function renderAttributeValueFormsAction(Request $request): Response
...
+            /** @var ProductAttributeInterface $attribute */
+            $attribute = $attributeValue->getAttribute();
+
+            $value = [
+                'id' => (string) $attributeValue->getId(),
+                'value' => $attributeValue->getValue(),
+                'label' => $attribute->getNameByLocaleCode($localeCode)
+            ];
+
+            if($attribute->isTranslatable()) {
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ProductBundle/Controller/ProductAttributeController.php:117]:
```suggestion
            if ($attribute->isTranslatable()) {
```

File: src/Sylius/Bundle/ProductBundle/Controller/ProductAttributeController.php [hunk 2]
```diff
@@ -79,6 +85,55 @@ public function renderAttributeValueFormsAction(Request $request): Response
...
+        /** @var string|null $productCode */
+        $productCode = $request->attributes->get('productCode');
+        Assert::notNull($productCode);
+
+        /** @var ProductRepositoryInterface|null $productRepository */
+        $productRepository = $this->get('sylius.repository.product');
+        Assert::notNull($productRepository);
+
+        /** @var ProductInterface $product */
+        $product = $productRepository->findOneByCode($productCode);
```
Comment (GSadee) [inline src/Sylius/Bundle/ProductBundle/Controller/ProductAttributeController.php:99]: Here should be checking if the product is not null I guess

File: src/Sylius/Component/Product/Model/Product.php [hunk 3]
```diff
@@ -229,7 +229,7 @@ public function hasAttributeByCodeAndLocale(string $attributeCode, ?string $loca
 
         foreach ($this->attributes as $attribute) {
             if ($attribute->getAttribute()->getCode() === $attributeCode
-                && $attribute->getLocaleCode() === $localeCode) {
+                && ($attribute->getLocaleCode() === $localeCode || null === $attribute->getLocaleCode())) {
                 return true;
```
Comment (GSadee) [inline src/Sylius/Component/Product/Model/Product.php:233]:
```suggestion
                && ($attribute->getLocaleCode() === $localeCode || null === $attribute->getLocaleCode())
            ) {
                return true;
```

**Comment by AdamKasp** [comment]
replaced by #12098

**Pull request closed without merging**